lib/https.js

```diff
@@ -6,6 +6,8 @@ const http = require('http');
 const util = require('util');
 const inherits = util.inherits;
 const debug = util.debuglog('https');
+const promises = require('internal/promises');
+const BuiltinPromise = promises.Promise;
```

Comment:
Inconsistent with http module. Should we prefer Promise over promises.Promise or the other way around?